app/views/devise/registrations/edit.html.erb

```diff
@@ -0,0 +1,59 @@
+<div class="centered-form">
+  <h2>Edit <%= resource_name.to_s.humanize %></h2>
+
+  <%= bootstrap_form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put, novalidate: true, class: "needs-validation" }) do |f| %>
+    <%= render "devise/shared/error_messages", resource: resource %>
+
+    <div class="mb-3">
+      <%= f.email_field :email, autofocus: true, autocomplete: "email", required: true, label: "Email" %>      
+    </div>
+
+    <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+      <div class="mb-3">
+        Currently waiting confirmation for: <%= resource.unconfirmed_email %>
+      </div>
+    <% end %>
+
+    <div class="mb-3">
+      <%= f.password_field :password, autocomplete: "new-password", label: "Password (leave blank if you don't want to change it)", required: false %>
+      <% if @minimum_password_length %>
+        <small class="form-text text-muted"><%= @minimum_password_length %> characters minimum</small>
+      <% end %>      
+    </div>
+
+    <div class="mb-3">
+      <%= f.password_field :password_confirmation, autocomplete: "new-password", label: "Password Confirmation", required: false %>      
+    </div>
+
+    <div class="mb-3">
+      <%= f.password_field :current_password, autocomplete: "current-password", required: true, label: "Current Password (we need your current password to confirm your changes)" %>      
+    </div>
+
+    <div class="d-flex justify-content-end mb-3">
+      <%= f.submit "Update", class: "btn btn-primary" %>
+    </div>
+  <% end %>
+
+  <h3>Cancel my account</h3>
+
+  <div>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete, class: "btn btn-danger" %></div>
+
+  <%= link_to "Back", :back, class: "btn btn-link" %>
+</div>
+
+<script>
```

Comment:
Are our assets (bootstrap) not loaded at this point? Would prefer to avoid inline js